### DISCUSSION TITLE
macbook pro(ARM M2) start banyan 0.6.1 alert  cannot execute binary file

### DISCUSSION THREAD

**Discussion by feixiangshenhua** [apache/skywalking#12303] [General]
I download banyandb 0.6.1 from https://dlcdn.apache.org/skywalking/banyandb/0.6.1/skywalking-banyandb-0.6.1-bin.tgz
then execute this command in **macbook（M2）**  and alert error. How to solve this problem?  Please help me,thanks.
**command：**  ./banyand-server-static standalone
**error info：** -bash: ./banyand-server-static: cannot execute binary file。

**Comment by feixiangshenhua** [comment]
This problem focus on **banyandb**, I am  Research **banyandb** is could replace **ES**。

**Reply by lujiajing1126** [reply]
I suppose this is not an universal binary

**Comment by wu-sheng** [comment]
Please follow this to build binaries

https://skywalking.apache.org/docs/skywalking-banyandb/latest/installation/binaries/#build-binaries

It should work on the Apple M chip. But for product env, it is better to build on Linux, or use the official docker images,

https://hub.docker.com/r/apache/skywalking-banyandb/tags

**Reply by feixiangshenhua** [reply]
Ok，I will verify by building

**Reply by feixiangshenhua** [reply]
It does't work, my go env is **GOARCH='arm64'**,and I build according to the document instructions  https://skywalking.apache.org/docs/skywalking-banyandb/latest/installation/binaries/#build-binaries 
It's also report error like this: 
**xiaoyun:bin xiaoyun$ ./banyand-server-static standalone -bash: ./banyand-server-static: cannot execute binary file**.
![banyanerror](https://github.com/apache/skywalking/assets/11437056/ca44b5f8-b2ae-4ab6-9543-e59e47e8e0e1)

**Comment by Superskyyy** [comment]
The binary is not for arm architecture yet, you have to build it yourself. Try this `export GOARCH=arm64` before building

**Reply by feixiangshenhua** [reply]
ok，I'm in trying.Thanks.

**Reply by feixiangshenhua** [reply]
It does't work, my go env is GOARCH='arm64',and I build according to the document instructions https://skywalking.apache.org/docs/skywalking-banyandb/latest/installation/binaries/#build-binaries
It's also report error like this:
xiaoyun:bin xiaoyun$ ./banyand-server-static standalone -bash: ./banyand-server-static: cannot execute binary file.

![banyanerror](https://github.com/apache/skywalking/assets/11437056/5f35a070-8d3f-423b-bb9a-16b545a61a4f)

what wrong with my operate?

**Reply by wu-sheng** [reply]
This should not be supported yet. For now, you may have to run amd64 image.

**Reply by Superskyyy** [reply]
we built banyandb on a aarch64 machine (Linux server EulerOS) and works, maybe Mac Mx is kind of different from other aarch64 systems?

**Reply by feixiangshenhua** [reply]
My mac is arm64, Can I run amd64 image，or only change my computer to amd64？
